# Yuxin Liao - Evil Geniuses x Genius League - Data Analyst

## Preclean data

In [1]:
import pandas as pd
df = pd.read_excel('social_data.xlsx')
print(df)

              Published Date   Account Account Type          Campaign Name  \
0    2023-03-31 19:55:43.192      CSGO      TWITTER                   N/A    
1    2023-03-31 19:49:35.243      CSGO      TWITTER                   N/A    
2    2023-03-31 19:49:33.535      CSGO      TWITTER                   N/A    
3    2023-03-31 19:49:31.583      CSGO      TWITTER                   N/A    
4    2023-03-31 19:43:42.590      CSGO      TWITTER  Community Engagement    
...                      ...       ...          ...                    ...   
3474 2023-01-03 10:35:12.652      CSGO      TWITTER                   N/A    
3475 2023-01-03 10:34:55.487      CSGO      TWITTER                   N/A    
3476 2023-01-03 10:34:38.631      CSGO      TWITTER                   N/A    
3477 2023-01-03 10:34:03.438      CSGO      TWITTER                   N/A    
3478 2023-01-01 14:59:04.000  General        FBPAGE                   N/A    

      Total Impressions  Total Engagements Media Type  
0      

#### Rename columns and remove NAs of Total Engagements

In [2]:
df = df.rename(columns={"Published Date": "Published_Date", "Account Type": "Account_Type", "Campaign Name": "Campaign_Name", "Total Impressions": "Total_Impressions", "Total Engagements": "Total_Engagements", "Media Type": "Media_Type"})

In [3]:
df.dropna(subset=['Total_Impressions'])

Published_Date   Account Account_Type          Campaign_Name  \
0    2023-03-31 19:55:43.192      CSGO      TWITTER                   N/A    
1    2023-03-31 19:49:35.243      CSGO      TWITTER                   N/A    
2    2023-03-31 19:49:33.535      CSGO      TWITTER                   N/A    
3    2023-03-31 19:49:31.583      CSGO      TWITTER                   N/A    
4    2023-03-31 19:43:42.590      CSGO      TWITTER  Community Engagement    
...                      ...       ...          ...                    ...   
3474 2023-01-03 10:35:12.652      CSGO      TWITTER                   N/A    
3475 2023-01-03 10:34:55.487      CSGO      TWITTER                   N/A    
3476 2023-01-03 10:34:38.631      CSGO      TWITTER                   N/A    
3477 2023-01-03 10:34:03.438      CSGO      TWITTER                   N/A    
3478 2023-01-01 14:59:04.000  General        FBPAGE                   N/A    

      Total_Impressions  Total_Engagements Media_Type  
0                   0.0                0.0       Text  
1                   0.0                0.0       Text  
2                   0.0                0.0       Text  
3                   0.0                0.0       Text  
4                9517.0             1215.0      Video  
...                 ...                ...        ...  
3474                0.0                0.0       Link  
3475                0.0                0.0       Link  
3476                0.0                0.0       Link  
3477             2116.0               42.0       Text  
3478                0.0                0.0      Photo  

[3479 rows x 7 columns]

## 1. What is the typical engagement rate we can expect? What’s the likelihood that we can achieve a 15% engagement rate?

In [4]:
df["Engagement_Rate"] = df["Total_Engagements"] / df["Total_Impressions"]

#### The typical engagement rate we can expect

In [5]:
rate = df["Engagement_Rate"].mean()*100
print(f'The typical engagement rate we can expect: {rate:0.2f}%.')

The typical engagement rate we can expect: 40.49%.


#### The likelihood that we can achieve a 15% engagement rate

In [6]:
Engagement_15_Amount = len(df[df["Engagement_Rate"]>=0.15])
Likelihood = Engagement_15_Amount/len(df)
print(f'The likelihood that we can achieve a 15% engagement rate: {Likelihood:0.2f}.')

The likelihood that we can achieve a 15% engagement rate: 0.06.


## 2. Does day of the week and time of posting affect engagement rates?

In [7]:
df["Published_Date"].dtypes

dtype('<M8[ns]')

In [8]:
df['day_of_week'] = df['Published_Date'].dt.day_name()
df['hour'] = df['Published_Date'].dt.hour
df

Published_Date   Account Account_Type          Campaign_Name  \
0    2023-03-31 19:55:43.192      CSGO      TWITTER                   N/A    
1    2023-03-31 19:49:35.243      CSGO      TWITTER                   N/A    
2    2023-03-31 19:49:33.535      CSGO      TWITTER                   N/A    
3    2023-03-31 19:49:31.583      CSGO      TWITTER                   N/A    
4    2023-03-31 19:43:42.590      CSGO      TWITTER  Community Engagement    
...                      ...       ...          ...                    ...   
3474 2023-01-03 10:35:12.652      CSGO      TWITTER                   N/A    
3475 2023-01-03 10:34:55.487      CSGO      TWITTER                   N/A    
3476 2023-01-03 10:34:38.631      CSGO      TWITTER                   N/A    
3477 2023-01-03 10:34:03.438      CSGO      TWITTER                   N/A    
3478 2023-01-01 14:59:04.000  General        FBPAGE                   N/A    

      Total_Impressions  Total_Engagements Media_Type  Engagement_Rate  \
0                   0.0                0.0       Text              NaN   
1                   0.0                0.0       Text              NaN   
2                   0.0                0.0       Text              NaN   
3                   0.0                0.0       Text              NaN   
4                9517.0             1215.0      Video         0.127666   
...                 ...                ...        ...              ...   
3474                0.0                0.0       Link              NaN   
3475                0.0                0.0       Link              NaN   
3476                0.0                0.0       Link              NaN   
3477             2116.0               42.0       Text         0.019849   
3478                0.0                0.0      Photo              NaN   

     day_of_week  hour  
0         Friday    19  
1         Friday    19  
2         Friday    19  
3         Friday    19  
4         Friday    19  
...          ...   ...  
3474     Tuesday    10  
3475     Tuesday    10  
3476     Tuesday    10  
3477     Tuesday    10  
3478      Sunday    14  

[3479 rows x 10 columns]

In [9]:
array2_1 = df["day_of_week"].unique()
array2_1

array(['Friday', 'Thursday', 'Wednesday', 'Tuesday', 'Monday', 'Sunday',
       'Saturday'], dtype=object)

In [10]:
for x in array2_1:
    df2_1 = df[df["day_of_week"] == x]
    rate2_1 = df2_1["Engagement_Rate"].mean()*100
    print('The average engagement rate of '+ x + f' we can expect: {rate2_1:0.2f}%.')

The average engagement rate of Friday we can expect: 166.04%.
The average engagement rate of Thursday we can expect: 9.84%.
The average engagement rate of Wednesday we can expect: 4.45%.
The average engagement rate of Tuesday we can expect: 6.54%.
The average engagement rate of Monday we can expect: 4.52%.
The average engagement rate of Sunday we can expect: 5.90%.
The average engagement rate of Saturday we can expect: 3.78%.


In [11]:
array2_2 = df["hour"].unique()
array2_2

array([19, 18, 17, 16, 15, 14, 13, 12, 11, 10,  9,  8,  7, 23, 21, 20, 22,
        6,  5,  4,  3,  0,  2])

In [12]:
for x in array2_2:
    df2_2 = df[df["hour"] == x]
    rate2_2 = df2_2["Engagement_Rate"].mean()*100
    print(f'The average engagement rate of {x} we can expect: {rate2_2:0.2f}%.')

The average engagement rate of 19 we can expect: 4.36%.
The average engagement rate of 18 we can expect: 4.56%.
The average engagement rate of 17 we can expect: 5.34%.
The average engagement rate of 16 we can expect: 5.32%.
The average engagement rate of 15 we can expect: 4.52%.
The average engagement rate of 14 we can expect: 5.05%.
The average engagement rate of 13 we can expect: 4.91%.
The average engagement rate of 12 we can expect: 396.70%.
The average engagement rate of 11 we can expect: 9.10%.
The average engagement rate of 10 we can expect: 4.76%.
The average engagement rate of 9 we can expect: 6.17%.
The average engagement rate of 8 we can expect: 7.41%.
The average engagement rate of 7 we can expect: 6.66%.
The average engagement rate of 23 we can expect: 2.86%.
The average engagement rate of 21 we can expect: 6.09%.
The average engagement rate of 20 we can expect: 4.79%.
The average engagement rate of 22 we can expect: 2.06%.
The average engagement rate of 6 we can expect: 2

In [13]:
#check for the significance of the day of the week in affecting the engagement rate
import scipy.stats as stats
result1 = stats.f_oneway(*[group['Engagement_Rate'] for name, group in df.groupby('day_of_week')])
result1

F_onewayResult(statistic=nan, pvalue=nan)

In [14]:
#check for the significance of the hour in affecting the engagement rate
import scipy.stats as stats
result2 = stats.f_oneway(*[group['Engagement_Rate'] for name, group in df.groupby('hour')])
result2

F_onewayResult(statistic=nan, pvalue=nan)

#### As we can see from the above result, Friday of the week and time at 12 has the highest engagement rates. We tried to conduct the anova test to see if the variables are significant in affecting the engagement rate. The p value is nan, which may result from the lack of enough statistics of the variables. 

In [15]:
#method 2
from scipy.stats import f_oneway
from scipy.stats import kruskal

grouped_data = df.groupby('day_of_week')['Engagement_Rate'].mean().reset_index()
grouped_data

f_statistic, p_value = f_oneway(*[df[df['day_of_week'] == day]['Engagement_Rate'] for day in df['day_of_week']])
p_value

nan

## 3. How are our game titles doing in terms of social performance? Is there a specific game we should focus more on or less?

In [16]:
array3 = df["Account"].unique()
array3

array(['CSGO', 'General ', 'DOTA2', 'Content Creators', 'Valorant',
       'General'], dtype=object)

In [17]:
df['Account'] = df['Account'].replace('General ', 'General')
array3 = df["Account"].unique()
array3

array(['CSGO', 'General', 'DOTA2', 'Content Creators', 'Valorant'],
      dtype=object)

In [18]:
for x in array3:
    df3 = df[df["Account"] == x]
    rate3 = df3["Engagement_Rate"].mean()*100
    print('The social performance in terms of average engagement rate of '+ x + f' we can expect: {rate3:0.2f}%.')

The social performance in terms of average engagement rate of CSGO we can expect: 4.13%.
The social performance in terms of average engagement rate of General we can expect: 59.93%.
The social performance in terms of average engagement rate of DOTA2 we can expect: 4.94%.
The social performance in terms of average engagement rate of Content Creators we can expect: 6.17%.
The social performance in terms of average engagement rate of Valorant we can expect: 5.34%.


#### The social performances of all five game titles are shown above in terms of their average engagement rate. Specifically, we should focus more on the games named "general" since it has the highest average engagement rate, surpassing the other four game titles. Game titles, including CSGO, DOTA2, Content Creators, and Valorant, have average engagement rates that range from 4% to 7%. And if we want to reduce focus on specific game titles, we should pay less focus on CSGO first since it has the lowest average engagement rate of 4.13%.

## 4. What media type performs the best?

In [19]:
array4 = df["Media_Type"].unique()
array4

array(['Text', 'Video', 'Photo', 'Link', 'Carousel', 'Mixed', 'Album'],
      dtype=object)

In [20]:
for x in array4:
    df4 = df[df["Media_Type"] == x]
    rate4 = df4["Engagement_Rate"].mean()*100
    print('The performance in terms of average engagement rate of '+ x + f' we can expect: {rate4:0.2f}%.')

The performance in terms of average engagement rate of Text we can expect: 4.08%.
The performance in terms of average engagement rate of Video we can expect: 5.35%.
The performance in terms of average engagement rate of Photo we can expect: 83.12%.
The performance in terms of average engagement rate of Link we can expect: 3.75%.
The performance in terms of average engagement rate of Carousel we can expect: 3.78%.
The performance in terms of average engagement rate of Mixed we can expect: 10.77%.
The performance in terms of average engagement rate of Album we can expect: 40.00%.


#### The media type of Photo performs best because it has the highest average engagement rate of 83.12%, which outperforms all other media types.

## 5. What is our best performing campaign?

In [21]:
array5 = df["Campaign_Name"].unique()
array5

array(['N/A ', 'Community Engagement ', 'Evergreen ', 'Evil Exhibited '],
      dtype=object)

In [22]:
for x in array5:
    if x != 'N/A ':
        df5 = df[df["Campaign_Name"] == x]
        rate5 = df5["Engagement_Rate"].mean()*100
        print('The performance in terms of average engagement rate of '+ x + f' we can expect: {rate5:0.2f}%.')

The performance in terms of average engagement rate of Community Engagement  we can expect: 5.77%.
The performance in terms of average engagement rate of Evergreen  we can expect: 3.51%.
The performance in terms of average engagement rate of Evil Exhibited  we can expect: 2.87%.


#### The campaign type of Community Engagement performs best because it has the highest average engagement rate of 5.77%, which outperforms all other campaign types.

## 6. Define out a posting strategy for our social channels based on your discoveries.

In [23]:
#check which account type generate higher engagement rate
array6 = df["Account_Type"].unique()
array6
for x in array6:
    df6 = df[df["Account_Type"] == x]
    rate6 = df6["Engagement_Rate"].mean()*100
    print('The performance in terms of average engagement rate of '+ x + f' we can expect: {rate6:0.2f}%.')

The performance in terms of average engagement rate of TWITTER we can expect: 3.83%.
The performance in terms of average engagement rate of FBPAGE we can expect: 237.20%.
The performance in terms of average engagement rate of INSTAGRAM we can expect: 1.08%.
The performance in terms of average engagement rate of TIKTOK_BUSINESS we can expect: 5.36%.
The performance in terms of average engagement rate of YOUTUBE we can expect: 4.47%.
The performance in terms of average engagement rate of LINKEDIN_COMPANY we can expect: 1.80%.


#### Based on the previous analysis, I would recommend to post more in the community engagement platforms using more photos. Also, posting on Friday around 12 PM will likely to engage more customers. The posting strategy is to fucus more on the general games, and using Facebook page will lead to more potential engagement.

## 7. What suggestions would you give to the social media team if they want to expand their presence (e.g. if our CSGO youtube channel is doing well should we expand to TikTok)?

To suggest to the social media team, I would need to conduct a comparative data analysis to compare the performance of the existing social media channels and the game types to be expanded. Generally, I will apply regression models, clustering algorithms, and collaborative filtering to identify similarities and predict the potential success of expanding to other platforms (e.g., TikTok) based on historical data. My recommendation to the social team will base on the integrated analysis outcome from all the below-mentioned analyses. 

Firstly, I will utilize predictive models, including time series forecasting and regression analysis, to predict audience growth, engagement rates, and revenue opportunities on TikTok. These models can help assess the scalability and viability of the expansion and guide resource allocation decisions. Also, I can use algorithms like K-means clustering and decision trees to identify similar target audiences on TikTok. Finally, by analyzing demographic, behavioral, and interest-based data, I can target users who resemble the existing engaged audience and increase the likelihood of success.

Additionally, I will apply NLP techniques to analyze comments, captions, and user-generated content on TikTok. Even though the language data is not provided in the current dataset, I will try to retrieve the online review and apply the NLP analysis. Using sentiment analysis and topic modeling will provide insights into audience preferences, content trends, and potential opportunities for the team to create relevant and engaging TikTok content.

Furthermore, I will investigate the social network analysis to understand the network structure and connections between the current Youtube channel and TikTok. By identifying the key influencers in the current channel and game type, I can further assess the potential for collaboration to expand the reach and engagement on TikTok.

Finally, I would implement A/B Testing: Design and execute A/B tests on TikTok to compare content strategies, game types, or suitable posting schedules on both platforms. The hypothesis testing and significance testing will help evaluate the performance of applying the same campaigns on TikTok based on data-driven insights and with controlled expenses. At the same time, I will closely monitor key metrics, such as impressions and engagements. I will set up real-time monitoring and data visualization dashboards to observe trends, identify anomalies, and make data-informed decisions to optimize the suggestion of whether to expand the successful campaign presence. 
